openpype/hosts/maya/api/lib.py

```diff
@@ -1971,6 +1979,35 @@ def _get_parent(node):
         if _id:
             return _id
 
+    if not history_only:
+        # Get siblings of same type
+        similar_nodes = cmds.listRelatives(parent,
+                                           type=node_type,
+                                           fullPath=True)
+        similar_nodes = cmds.ls(similar_nodes, exactType=node_type, long=True)
+
+        # Exclude itself
+        similar_nodes = [x for x in similar_nodes if x != node]
+
+        first_id = None
+        found_node = None
+        for similar_node in similar_nodes:
+            # Check if "intermediate object"
+            if cmds.getAttr(similar_node + ".intermediateObject"):
+                _id = get_id(similar_node)
+                if _id:
+                    # Check if already found an id
+                    if first_id:
+                        log.warning(("Found more than 1 matching intermediate"
+                                     " shape for '{}'. Using id of first"
+                                     " found: '{}'".format(node, found_node)))
+                        break
+                    first_id = _id
+                    found_node = similar_node
+
+        return first_id
```

Comment:
This logic has the downside to also log a warning if both shapes found had the same id - which would still be 'safe'.
We could do `if first_id != _id:` or alike.

Or here's an alternative approach:

```python
# pseudocode
from collections import OrderedDict
...
        
        # Get all unique ids from siblings in order since
        # we consistently take the first one found
        sibling_ids = OrderedDict()
        for similar_node in similar_nodes:
            # Check if "intermediate object"
            if not cmds.getAttr(similar_node + ".intermediateObject"):
                continue
                
            _id = get_id(similar_node)
            if not _id:
                continue
                
            if _id in sibling_ids:
                sibling_ids[_id] = [similar_node]
            else:
                sibling_ids[_id].append(similar_node)
                    
        if sibling_ids:
            # Log a warning if we've found multiple unique ids
            first_id, found_nodes = next(iter(sibling_ids.items()))
            if len(sibling_ids) > 1:
                log.warning(("Found more than 1 intermediate shape with unique"
                             " id: for '{}'. Using id of first"
                             " found: '{}'".format(node, found_nodes[0])))
                break
               
            return first_id
```

Reply:
Ah yes, I didn't know the case with same IDs was valid/possible.
In that case, your alternative approach seems good, I will make the change.

I will check that ok, and update the changes.

Reply:
> (and the user needs to check by themself anyway, as the tool will not fix it).

Actually, the tool does fix it if you right click -> repair. Anyway, agreed. It's something left to the user to debug.